qiskit/dagcircuit/dagdependency_v2.py

```diff
@@ -0,0 +1,688 @@
+                    )
+                    for predecessor_id in predecessor_ids:
+                        reachable[predecessor_id] = False
+            else:
+                # If prev_node cannot reach max_node, then none of its predecessors can
+                # reach max_node either.
+                predecessor_ids = sorted(
+                    [node._node_id for node in self.predecessors(self._get_node(prev_node_id))]
+                )
+                for predecessor_id in predecessor_ids:
+                    reachable[predecessor_id] = False
+
+    def _get_node(self, node_id):
+        """
+        Args:
+            node_id (int): label of considered node.
+
+        Returns:
+            node: corresponding to the label.
+        """
+        return self._multi_graph.get_node_data(node_id)
+
+    def remove_all_ops_named(self, opname):
+        """Remove all operation nodes with the given name."""
+        for n in self.named_nodes(opname):
+            self.remove_op_node(n)
+
+    def named_nodes(self, *names):
+        """Get the set of "op" nodes with the given name."""
+        named_nodes = []
+        for node in self._multi_graph.nodes():
+            if node.op.name in names:
+                named_nodes.append(node)
+        return named_nodes
+
+    def _increment_op(self, op):
+        if op.name in self._op_names:
+            self._op_names[op.name] += 1
+        else:
+            self._op_names[op.name] = 1
+
+    def _decrement_op(self, op):
+        if self._op_names[op.name] == 1:
+            del self._op_names[op.name]
+        else:
+            self._op_names[op.name] -= 1
+
+    def count_ops(self):
+        """Count the occurrences of operation names."""
+        return self._op_names
+
+    def remove_op_node(self, node):
+        """Remove an operation node n.
+
+        Add edges from predecessors to successors.
+        """
+        self._multi_graph.remove_node_retain_edges(
+            node._node_id, use_outgoing=False, condition=lambda edge1, edge2: edge1 == edge2
+        )
+        self._decrement_op(node.op)
+
+    def op_nodes(self):
+        """
+        Returns:
+            generator(dict): iterator over all the nodes.
+        """
+        return iter(self._multi_graph.nodes())
+
+    def topological_nodes(self, key=None) -> Generator[DAGOpNode, Any, Any]:
+        """
+        Yield nodes in topological order.
+
+        Args:
+            key (Callable): A callable which will take a DAGNode object and
+                return a string sort key. If not specified the
+                :attr:`~qiskit.dagcircuit.DAGNode.sort_key` attribute will be
+                used as the sort key for each node.
+
+        Returns:
+            generator(DAGOpNode): node in topological order
+        """
+
+        def _key(x):
+            return x.sort_key
+
+        if key is None:
+            key = _key
+
+        return iter(rx.lexicographical_topological_sort(self._multi_graph, key=_key))
```

Comment:
```suggestion
        return iter(rx.lexicographical_topological_sort(self._multi_graph, key=key))
```
I just stumbled upon this typo while working on something else. :-)